### ISSUE TITLE
Documentation for I/O Pin classes is incorrect for digital PWM  support

### ISSUE THREAD

**Issue by MrYsLab** [bbcmicrobit/micropython#487]
write_analog, set_analog_period and set_analog_period_microseconds should be moved from [class microbit.MicroBitAnalogDigitalPin](http://microbit-micropython.readthedocs.io/en/latest/pin.html#microbit.MicroBitAnalogDigitalPin) to [class microbit.MicroBitDigitalPin](http://microbit-micropython.readthedocs.io/en/latest/pin.html#microbit.MicroBitDigitalPin) so that the hierarchy flows correctly.

In addition, set_pull should be added to class microbit.MicroBitDigitalPin

**Comment by deshipu** [comment]
Would you be willing to make those changes and submit a pull request? We can help you with that.

**Comment by MrYsLab** [comment]
No problem. I will submit ASAP (probably within a couple of days). I am familiar with pull requests but will ask for help if I run into any issues.

**Comment by microbit-carlos** [comment]
@MrYsLab We are working through the documentation at the moment, are you still interested to create a PR for this?

**Comment by MrYsLab** [comment]
Yes, PR #488 was submitted to address the issue.

**Comment by MrYsLab** [comment]
@microbit-carlos Going over this, it may not be clear as to my rationale. When I created [pseudo-microbit](https://github.com/MrYsLab/pseudo-microbit]) I tried following the documentation to create the pin class so that PyCharm could do auto-completion. Following the documentation as it is currently written, when I created the classes, things did not work. I had to modify the pin class definitions to match my request. There is also some clean-up reorganization (IMHO) to make things easier to understand.
